mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -1024,6 +1192,168 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
                                                ArrayRef<Value>(embeddedInput),
                                                transformedInputAttrs);
 
+    auto gemmBPad = gemmB;
+    bool isInputPad = false;
+    // input padding start
+    if (convOpType == miopen::ConvOpType::Conv2DOpType && gemmK_extra > 0) {
+      llvm::SmallVector<int64_t, 3> paddingInputShape;
```

Comment:
same as above, comment is needed to explain the general idea behind the algorithm.